tests/unit/backend/layers/business/test_business.py

```diff
@@ -0,0 +1,778 @@
+        }
+
+        self.business_logic.update_collection_version(new_version.version_id, body, self.user_info)
+        self.business_logic.publish_collection_version(new_version.version_id, self.user_info)
+
+        new_version_from_db = self.database_provider.get_collection_version(new_version.version_id)
+        self.assertIsNone(new_version_from_db.published_at)
+        self.assertEqual(new_version.collection_id, new_version_from_db.collection_id)
+        self.assertEqual(new_version.version_id, new_version_from_db.version_id)
+        self.assertEqual(new_version.datasets, new_version_from_db.datasets)
+
+    def test_publish_version_with_removed_dataset_ok(self):
+        """
+        Publishing a version with a dataset removed should succeed
+        """
+        public_collection = self.initialize_public_collection()
+        new_version = self.business_logic.create_collection_version(public_collection.collection_id, self.user_info)
+
+        dataset_version_to_remove = new_version.datasets[0].version_id
+        dataset_version_to_keep = new_version.datasets[1].version_id
+
+        self.business_logic.delete_dataset(dataset_version_to_remove)
+
+        # The new version should have only one dataset (before publishing)
+        version_from_db = self.database_provider.get_collection_version(new_version.version_id)
+        self.assertEqual(1, len(version_from_db.datasets))
+
+        # The original version should still have two datasets (before publishing)
+        version_from_db = self.database_provider.get_collection_version(public_collection.version_id)
+        self.assertEqual(2, len(version_from_db.datasets))
+
+        # Get collection retrieves the original version (with two datasets)
+        version = self.business_logic.get_collection(public_collection.collection_id, self.user_info)
+        self.assertEqual(version.version_id, public_collection.version_id)
+        self.assertEqual(2, len(version.datasets))
+
+        self.business_logic.publish_collection_version(new_version.version_id, self.user_info)
+
+        # The new version should have only one dataset (after publishing)
+        version_from_db = self.database_provider.get_collection_version(new_version.version_id)
+        self.assertEqual(1, len(version_from_db.datasets))
+
+        # The old version should still have two datasets (after publishing)
+        version_from_db = self.database_provider.get_collection_version(public_collection.version_id)
+        self.assertEqual(2, len(version_from_db.datasets))
+
+        # Get collection retrieves the new version (with one datasets)
+        version = self.business_logic.get_collection(public_collection.collection_id, self.user_info)
+        self.assertEqual(version.version_id, new_version.version_id)
+        self.assertEqual(1, len(version.datasets))
+        self.assertEqual(version.datasets[0].version_id, dataset_version_to_keep)
+
+
+    def test_publish_version_with_added_dataset_ok(self):
```

Comment:
All of the `test_publish_version_with_*` tests are good to have, as a black box test. In practice, we're no longer doing all of the sophisticated copying and deleting of records that occurs in the current system. So from a white box testing perspective, having all of these tests doesn't add much, right? A "publish" operation is just a simple collection version attribute and collection mapping update, right?

I'm not suggesting we should remove them, but rather trying to ensure my understanding is correct.

Reply:
You're correct, but I still think these don't hurt (as you specified).